mlock and password do not build on AIX (GOOS=aix)

## Description

The `mlock` and `password` modules fail to compile for `GOOS=aix GOARCH=ppc64`:

```
mlock/mlock.go:17:9: undefined: lockMemory
password/password.go:34:23: undefined: read
```

Neither `mlock_unix.go` nor `password_unix.go` includes `aix` in its build constraint, so the platform falls through to no implementation at all (mlock does not even reach `mlock_unavail.go`, whose constraint also excludes aix).

AIX supports both underlying facilities: `syscall.Mlockall` / `plock` semantics used by `mlock_unix.go`, and the termios reads used by `password_unix.go` both compile and run on `aix/ppc64` with the stock Go toolchain.

## Use case

Building Vault's CLI/Agent for AIX on IBM Power. The IBM Power integration team has been carrying this as a one-line fork patch (adding `aix` to the two build constraints) and it has been validated in their AIX builds.

## Proposed fix

Add `aix` to the build constraints of `mlock/mlock_unix.go` and `password/password_unix.go`. PR to follow.

--- a/mlock/mlock_unix.go
+++ b/mlock/mlock_unix.go
@@ -1,7 +1,8 @@
 // Copyright IBM Corp. 2020, 2025
 // SPDX-License-Identifier: MPL-2.0
 
-// +build dragonfly freebsd linux openbsd solaris
+//go:build aix || dragonfly || freebsd || linux || openbsd || solaris
+// +build aix dragonfly freebsd linux openbsd solaris
 
 package mlock
 

--- a/password/password_unix.go
+++ b/password/password_unix.go
@@ -1,8 +1,8 @@
 // Copyright IBM Corp. 2020, 2025
 // SPDX-License-Identifier: MPL-2.0
 
-//go:build linux || darwin || freebsd || netbsd || openbsd || dragonfly || js
-// +build linux darwin freebsd netbsd openbsd dragonfly js
+//go:build linux || darwin || freebsd || netbsd || openbsd || dragonfly || js || aix
+// +build linux darwin freebsd netbsd openbsd dragonfly js aix
 
 package password
 

